Framework/CurveFitting/src/FuncMinimizers/LevenbergMarquardtMinimizer.cpp

```diff
@@ -114,6 +114,11 @@ bool LevenbergMarquardtMinimizer::iterate(size_t) {
 
   if (retVal && retVal != GSL_CONTINUE) {
     m_errorString = gsl_strerror(retVal);
+    if (m_errorString == "cannot reach the specified tolerance in F") {
```

Comment:
Are you sure it's not possible to identify these errors using [`gsl_errno.h`](https://github.com/wdavidw/node-gsl/blob/master/deps/gsl-1.14/err/gsl_errno.h#L71-L72)?

Reply:
it might be. However, I think this is more transparent as we have the original error message explicitly stated

Reply:
True, but it is also more brittle/likely to silently break.  If they wanted you to depend on the contents of the error messages then they probably wouldn't provide an errno as well. I notice you're checking for `GSL_CONTINUE` and not `"iteration has not converged"` above.

I would suggest that ```c++
if(cannotReachSpecifiedToleranceInF(retVal)) {
  // ...

Reply:
where are getting the function from?